lib/json/common.rb

```diff
@@ -613,26 +613,42 @@ class << self
   # Output:
   #   {"foo":[0,1],"bar":{"baz":2,"bat":3},"bam":"bad"}
   def dump(obj, anIO = nil, limit = nil, kwargs = nil)
```

Comment:
One idea (which I learned from @nobu IIRC) to optimize the most common case of no extra arguments are passed is:
```ruby
def dump(obj, anIO = (no_args_set = true; nil), limit = nil, kwargs = nil)
  unless no_args_set
    ...
  end

  opts = JSON.dump_default_options
  ...
```

Reply:
Nice trick.